packages/task/src/browser/task-service.ts

```diff
@@ -153,6 +160,19 @@ export class TaskService implements TaskConfigurationClient {
         return this.providedTaskConfigurations.getTasks();
     }
 
+    getRecentTasks(): Promise<TaskConfiguration[]> {
+        return this.storageService.getData<Array<TaskConfiguration>>(RECENT_TASK_STORAGE_KEY, []);
+    }
+
+    storeRecentTask(task: TaskConfiguration): Promise<void> {
+        const stored = this.recentTasks.some(recent => TaskConfiguration.equals(recent, task));
+        if (!stored) {
+            this.recentTasks.push(task);
+            return this.storageService.setData<Array<TaskConfiguration>>(RECENT_TASK_STORAGE_KEY, this.recentTasks);
```

Comment:
It should be enough to save recent tasks to local storage only once on `TaskFrontendContribution.onStop`.

`TaskService` does not need to know about `StorageService`, it can keep recent tasks and be populated from `TaskFrontendContribution` on start up.

I would call a method  `pushRecentTask` or `addRecentTask` and it should only mutate `recentTasks`.

Reply:
moved the storageservice part to TaskFrontendContribution.

Thank you for the review !